pkgs/development/tools/build-managers/bazel/bazel_7/default.nix

```diff
@@ -117,8 +100,126 @@ let
       unzip
       which
       zip
+      runJdk
+      makeWrapper
     ];
 
+  # Bootstrap an existing Bazel so we can vendor deps with vendor mode
```

Comment:
Were we using a binary bootstrap before, or is this a new change? I thought that Bazel always required itself to bootstrap, so it’d help me if you could explain the differences from the previous situation.

Currently it seems like this would restrict Bazel to `x86_64-linux` only, which would be unfortunate. Would keeping the old Bazel around to build this one provide us a source‐based bootstrap path, or would we just be using a different binary bootstrap?

Reply:
Bazel does require itself to bootstrap as part of its own build, but we now need a version of Bazel 7.3 that we can use prior to building Bazel 7.3 in order to be able to run `bazel vendor` for the FOD that fetches the deps.

We should be able to bootstrap all combinations of darwin/linux and x86_64/arm64. We can certainly do that for linux, though I don't have an arm64-linux machine readily available. My understanding with the darwin derivation is that it builds but is essentially unusable, but I don't have the experience to confirm this. I'd be okay with delisting darwin support as long as people with this experience can verify that understanding.

Reply:
One thing I'm not certain about: we need the bootstrap without an existing Bazel >= 7.3, but is there a way to get rid of it once we've done an initial bootstrap?

Reply:
This now should work for all supported platforms, though I haven't tested it on all of them yet.

Reply:
This now should work for all supported platforms, though I haven't tested it on all of them yet.

Reply:
Thanks, I understand better now. Am I correct in thinking that previously we did not need a binary blob bootstrap Bazel and now we do, and that there is no way to use the old from‐source Bazel to bootstrap the new one? That is unfortunate, but not something we really have control over. You should set at least `meta.sourceProvenance` on the bootstrap Bazel accordingly.

Reply:
That is correct, at least for the approach I've taken to vendor the deps. However, once we have a working Bazel 7.3 cached, is there a way for us to use that instead? It's not obvious to me how, but I figured there might be a way to eliminate the external bootstrap after the fact.

> You should set at least meta.sourceProvenance on the bootstrap Bazel accordingly.

done https://github.com/NixOS/nixpkgs/pull/338264/commits/6338bd358ef6d566422e613c31a57a7e568e6960

Reply:
We have the bootstrap tools tarballs, but those are for things that are required to rebuild the standard environment. I think it wouldn’t make much sense in this case, because the provenance would be essentially the same: we’re trusting the opaque blob to produce a new opaque blob for us, so there’s no point pretending that “laundering” it through Hydra produces something we can confidently say was bootstrapped cleanly from the original source code.